src/core/main.c

```diff
@@ -1382,6 +1383,39 @@ static int os_release_status(void) {
         return 0;
 }
 
+static int setup_os_release(RuntimeScope scope) {
+        _cleanup_free_ char *os_release_dst = NULL;
+        const char *os_release_src;
+        int r;
+
+        if (access("/etc/os-release", F_OK) < 0) {
+                if (errno != ENOENT)
+                        log_debug_errno(errno, "Failed to check if /etc/os-release exists, ignoring: %m");
+
+                os_release_src = "/usr/lib/os-release";
+        } else
+                os_release_src = "/etc/os-release";
+
+        if (scope == RUNTIME_SCOPE_SYSTEM) {
+                os_release_dst = strdup("/run/systemd/propagate/os-release");
+                if (!os_release_dst)
+                        return log_oom_debug();
+        } else {
+                if (asprintf(&os_release_dst, "/run/user/" UID_FMT "/systemd/propagate/os-release", geteuid()) < 0)
+                        return log_oom_debug();
+        }
+
+        r = mkdir_parents_label(os_release_dst, 0755);
+        if (r < 0 && r != -EEXIST)
+                return log_error_errno(r, "Failed to create os-release propagate directory, ignoring: %m");
+
+        r = copy_file(os_release_src, os_release_dst, /* open_flags= */ 0, 0644, COPY_MAC_CREATE);
+        if (r < 0)
+                return log_error_errno(r, "Failed to copy os-release to propagate directory, ignoring: %m");
```

Comment:
Both of those messages would be much easier to grok if the actual path was given.
Why not just say: `"Failed to create %s, ignoring", os_release_dst` ?

(This is a general anti-pattern in our code. We invent plain-English descriptions of concepts in error messages, and it's sometimes very hard to map those descriptions to anything useful.)